operator/v1/types.go

```diff
@@ -45,7 +46,6 @@ var (
 // inside of the Spec struct for your particular operator.
 type OperatorSpec struct {
 	// managementState indicates whether and how the operator should manage the component
-	// +kubebuilder:validation:Pattern=^(Managed|Unmanaged|Force|Removed)$
```

Comment:
The fact this was working for anyone makes me somewhat worried that moving it will break an older generator version which may have supported this. I'm not even sure how we can test who would be affected? Is it ever safe to change these annotations once set given the metadata schema itself isn't versioned?

Reply:
Is this move necessary?

Reply:
https://github.com/openshift/cluster-dns-operator/pull/113
https://github.com/openshift/cluster-config-operator/pull/65

Looks like this change should. be okay for at least a couple of existing generators which cover this type.

Reply:
@sttts yeah, latest controller-tools crd generator enforces Pattern be declared on the alias type in this context.

Reply:
@ironcladlou is that only for pattern?

Reply:
@sttts Been thinking about this some more and digging through the generator source again which led me to something I didn't see before. Adding a type specifier is possible:

```
// +kubebuilder:validation:Type=string
// +kubebuilder:validation:Pattern=^(Managed|Unmanaged|Force|Removed)$
ManagementState ManagementState `json:"managementState"`
```

However, while doing so satisfies the generator, `Pattern` is missing from the resulting generated schema. I could be using it wrong.

Reply:
Spoke with @sttts and @directxman12 — looks like in this particular case an `enum` or `pattern` on the type is appropriate given the static value set, but for other cases we'll need field-level support (which already works for `string` but may have a bug with alias types which we can fix if true).

Reply:
And, `Type` eating `Pattern` may be another bug to chase down.